File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/36_compito_java_2020_01_16/Ipermercato.java
public class Ipermercato extends Supermercato {
	private float superficie;
	private int numeroAddetti;
	private String nomeCentroCommerciale;
	private int numeroVisite;

	public Ipermercato(int codice, String tipo, String indirizzo, float superficie, int numeroAddetti,
			String nomeCentroCommerciale) {
		super(codice, tipo, indirizzo);
		this.superficie = superficie;
		this.numeroAddetti = numeroAddetti;
		this.nomeCentroCommerciale = nomeCentroCommerciale;
		numeroVisite = 0;
	}

	public void addvisite() {
		numeroVisite++;
	}

	public int getNumeroVisite() {
		return numeroVisite;
	}

	public String toString() {
		return codice + "\t" + tipo + "\t" + indirizzo + "\t-\t-\t" + superficie + "\t" + numeroAddetti + "\t"
				+ nomeCentroCommerciale;
	}
}

File: Programming-languages-and-laboratory/Python/Lessons/25_20221128/36_soluzione/Ipermercato.py
from Supermercato import Supermercato


class Ipermercato(Supermercato):
    def __init__(self, codice, indirizzo, superficie, addetti, centro):
        super().__init__(codice, indirizzo, superficie, addetti, "iper")
        self._centro = centro

            
    def toString(self):
        return super().toString() + "-\t-\t"+str(self._superficie) + "\t" + str(self._addetti) + "\t" + self._centro